simphony/engine/tests/test_engine_metadata.py

```diff
@@ -0,0 +1,134 @@
+"""Tests regarding loading engine's metadata."""
+import sys
+import unittest
+
+import simphony.engine as engine_api
+from simphony.engine import ABCEngineExtension, EngineInterface
+from simphony.engine.extension import EngineManager, EngineManagerException
+from simphony.engine.extension import EngineFeatureMetadata, EngineMetadata
+
+
+class _Example1(ABCEngineExtension):
+    def get_supported_engines(self):
+        example_engine = \
+            self.create_engine_metadata('EXAMPLE1',
+                                        None,
+                                        [EngineInterface.Internal,
+                                         EngineInterface.FileIO])
+
+        return [example_engine]
+
+    def create_wrapper(self, cuds, engine_name, engine_interface):
+        if engine_name == 'EXAMPLE1':
+            pass
+        else:
+            raise Exception('Only EXAMPLE1 engine is supported. '
+                            'Unsupported eninge: %s', engine_name)
+
+
+def get_example_engine_extension():
+    class _Example2(ABCEngineExtension):
+        def get_supported_engines(self):
+            example_engine = \
+                self.create_engine_metadata('EXAMPLE2',
+                                            None,
+                                            [EngineInterface.Internal,
+                                             EngineInterface.FileIO])
+
+            return [example_engine]
+
+        def create_wrapper(self, cuds, engine_name, engine_interface):
+            if engine_name == 'EXAMPLE2':
+                pass
+            else:
+                raise Exception('Only EXAMPLE2 engine is supported. '
+                                'Unsupported eninge: %s', engine_name)
```

Comment:
> Exception("blahblah %s" % engine_name) ? (and there is another one on L26)

I assume you mean why I have hard-coded these dummy names. Those engines are there only for sake of testing. I do not rely on the other engines (like lammps) to be loaded, that should be part of each specific wrapper's tests.